[BUG] Table or view may not be selected while mirroring.

## Bug

### Describe the problem

Table or view may not be selected while mirroring.

#### Steps to reproduce

1. Have table mirrored using library regularly (parameters don't matter), e.g. named `nakl.dbo__tab_zakaz`
2. Selecting while mirroring may give error `Caused by: org.apache.spark.sql.AnalysisException: Table or view not found: nakl.dbo__tab_zakaz; line 19 pos 15;`

#### Observed results

Table or view not found.

#### Expected results

Table or view should exists always.

### Environment information

* Mirroring library version: 1.0.6
* Spark version: 3.3.2
* Scala version: 2.12.10

### Willingness to contribute

The Delta Flow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the Delta Flow code base?

- [x] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the Delta Flow community.
- [ ] No. I cannot contribute a bug fix at this time.

--- a/src/main/scala/mirroring/builders/SqlBuilder.scala
+++ b/src/main/scala/mirroring/builders/SqlBuilder.scala
@@ -45,16 +45,12 @@ object SqlBuilder extends LogSupport {
     s" AND $dtFlt >= '${startDate}T00:00:00.000' AND $dtFlt < '${endDate}T00:00:00.000'"
   }
 
-  def buildDropTableSQL(db: String, tab: String): String = {
-    s"DROP TABLE IF EXISTS `$db`.`$tab`;"
-  }
-
   def buildCreateTableSQL(
       db: String,
       tab: String,
       dataPath: String
   ): String = {
-    s"""CREATE EXTERNAL TABLE `$db`.`$tab`
+    s"""CREATE EXTERNAL TABLE IF NOT EXISTS `$db`.`$tab`
     |USING DELTA
     |LOCATION '$dataPath';
     """.stripMargin

--- a/src/main/scala/mirroring/services/SqlService.scala
+++ b/src/main/scala/mirroring/services/SqlService.scala
@@ -29,25 +29,15 @@ object SqlService extends LogSupport {
 
     val createDbSQL =
       SqlBuilder.buildCreateDbSQL(config.hiveDb, config.hiveDbLocation)
-    val dropTableSQL =
-      SqlBuilder.buildDropTableSQL(config.hiveDb, config.targetTableName)
     val createTableSQL = SqlBuilder.buildCreateTableSQL(
       config.hiveDb,
       config.targetTableName,
       config.pathToSave
     )
-    val alterTableSQL = SqlBuilder.buildAlterTableSQL(
-      config.hiveDb,
-      config.targetTableName,
-      config.logRetentionDuration,
-      config.deletedFileRetentionDuration
-    )
 
     if (DeltaTable.isDeltaTable(spark, config.pathToSave)) {
       logger.info(s"Running SQL: $createDbSQL")
       spark.sql(createDbSQL)
-      logger.info(s"Running SQL: $dropTableSQL")
-      spark.sql(dropTableSQL)
       logger.info(s"Running SQL: $createTableSQL")
       spark.sql(createTableSQL)
 
@@ -70,6 +60,12 @@ object SqlService extends LogSupport {
         !logRetentionDuration.equals(config.logRetentionDuration) || !deletedFileRetentionDuration
           .equals(config.deletedFileRetentionDuration)
       ) {
+        val alterTableSQL = SqlBuilder.buildAlterTableSQL(
+          config.hiveDb,
+          config.targetTableName,
+          config.logRetentionDuration,
+          config.deletedFileRetentionDuration
+        )
         spark.sql(alterTableSQL)
       }
     }

--- a/src/test/scala/mirroring/SqlBuilderSuite.scala
+++ b/src/test/scala/mirroring/SqlBuilderSuite.scala
@@ -42,13 +42,6 @@ class SqlBuilderSuite extends AnyFunSuite {
     )
   }
 
-  test("buildDropTableSQL should create DROP TABLE statement") {
-    val table  = "Test1Table"
-    val db     = "default"
-    val result = SqlBuilder.buildDropTableSQL(db, table)
-    assert(result == "DROP TABLE IF EXISTS `default`.`Test1Table`;")
-  }
-
   test(
     "buildCreateTableSQL should generate CREATE EXTERNAL TABLE statement"
   ) {
@@ -61,7 +54,7 @@ class SqlBuilderSuite extends AnyFunSuite {
       dataPath
     )
     val expectedResult =
-      s"""CREATE EXTERNAL TABLE `default`.`Test1Table`
+      s"""CREATE EXTERNAL TABLE IF NOT EXISTS `default`.`Test1Table`
        |USING DELTA
        |LOCATION 's3a://bucket/folder';
     """.stripMargin

